src/aead/chacha20_poly1305.rs

```diff
@@ -278,6 +279,89 @@ mod tests {
                     (CRYPTO_POLY1305_STATE_LEN + 255) / 256);
     }
 
+    // This verifies the encryption functionality provided by ChaCha20_ctr32
+    // is successful when either computed on distinct input/output buffers,
+    // or in-place.
+    //
+    // On certain architectures (x86 and some branches of ARM), the in-place
+    // operation only succeeds when the input/output buffers are exactly
+    // overlapping due to the assembly code. Hence we do not test offsets for 
+    // x86 and ARM architectures.
+    //
+    // See the body of |open| for more information and the workaround.
+    #[test]
+    pub fn chacha20_tests() {
+        test::from_file("src/chacha_tests.txt", |section, test_case| {
```

Comment:
Sure. Should really move all src/*_tests.txt where corresponding module folders exist.